src/microbe_stage/Microbe.cs

```diff
@@ -1355,6 +1366,26 @@ public override void _IntegrateForces(PhysicsDirectBodyState physicsState)
         physicsState.Transform = GetNewPhysicsRotation(physicsState.Transform);
     }
 
+    public void ToggleSprint()
+    {
+        RequestSprint(!isSprinting);
+    }
+
+    public void RequestSprint(bool state)
+    {
+        if (state)
+        {
+            if (Compounds.GetCompoundAmount(atp) > Compounds.Capacity * Constants.MIN_ATP_TO_START_SPRINT)
+            {
+                isSprinting = true;
+            }
+        }
+        else
+        {
+            isSprinting = false;
+        }
+    }
+
```

Comment:
I think these should be represented by a property instead of having explicit setters like this. Similarly to how the microbe state transitions have extra logic implemented as a property.